### ISSUE TITLE
App crashes, if share button is pressed

### ISSUE THREAD

**Issue by d137** [nextcloud/android#5615]
### Steps to reproduce
1. Nginx (1.17.9) with reverse proxy
2. ssl_protocols TLSv1.3; ssl_ciphers HIGH:!aNULL:!MD5;
3. http2 enabled
4. Open App, hit share button on any file

### Expected behaviour
- Hit share button
- Get share view
- Share file

### Actual behaviour
- Hit share button
- Opens view where I can see the share view for two seconds
- App crashes with exception

### Environment data
Android version: 10

Device model: star2lte (Samsung S9+)

Stock or customized system: Stock

Nextcloud app version: 30100190

Nextcloud server version: 18.0.0

Nginx reverse proxy: 1.17.9

#### Android App Exception log
```
************ CAUSE OF ERROR ************

javax.net.ssl.SSLHandshakeException: Handshake failed
at com.google.android.gms.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(:com.google.android.gms@200616037@20.06.16 (120400-296104215):38)
at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.java:320)
at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.java:284)
at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:169)
at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
at okhttp3.RealCall.execute(RealCall.java:93)
at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:126)
at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:84)
at com.owncloud.android.lib.resources.activities.GetActivitiesRemoteOperation.run(GetActivitiesRemoteOperation.java:131)
at com.nextcloud.common.NextcloudClient.execute(NextcloudClient.kt:80)
at com.owncloud.android.ui.fragment.FileDetailActivitiesFragment.lambda$fetchAndSetData$5$FileDetailActivitiesFragment(FileDetailActivitiesFragment.java:331)
at com.owncloud.android.ui.fragment.-$$Lambda$FileDetailActivitiesFragment$N7POe6TZvoGcuV2qP-niPxY_7t4.run(Unknown Source:12)
at java.lang.Thread.run(Thread.java:919)
Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x758fbf6448: Failure in SSL library, usually a protocol error
error:1000042e:SSL routines:OPENSSL_internal:TLSV1_ALERT_PROTOCOL_VERSION (third_party/openssl/boringssl/src/ssl/tls_record.cc:592 0x762f6a1908:0x00000001)
at com.google.android.gms.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
at com.google.android.gms.org.conscrypt.NativeSsl.doHandshake(:com.google.android.gms@200616037@20.06.16 (120400-296104215):6)
at com.google.android.gms.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(:com.google.android.gms@200616037@20.06.16 (120400-296104215):14)
... 26 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: star2lte
Model: SM-G965F
Id: QP1A.190711.020
Product: star2ltexx

************ FIRMWARE ************
SDK: 29
Release: 10
Incremental: G965FXXS7DTAD
```

#### Web server error log
```
Nothing shown, not hitting log area
```

#### Nextcloud log (data/nextcloud.log)
```
Nothing shown, not hitting log area
```

**Comment by diezorg** [comment]
Duplicate of #5577

**Comment by diezorg** [comment]
Installation with TLS 1.2 and http2 with same Android App seems fine.

**Comment by diezorg** [comment]
Switched back to TLS 1.2, app works.

**Comment by AndyScherzinger** [comment]
@diezorg thanks for the feedback, so closing in favor

**Comment by tobiasKaminsky** [comment]
@d137 so this crashes with TLS1.3, but not with 1.2?

**Comment by tobiasKaminsky** [comment]
Can you create us a test account, test if the problem occurs also there and if so send the credentials to tobias at nextcloud dot com with a reference to this issue?

**Comment by d137** [comment]
@tobiasKaminsky Yes nginx with TLS1.3 -> crash. With TLS1.2 it works without crash.
I created and mailed you the test account.

**Comment by d137** [comment]
Seems to be fixed in latest android release 3.11.0 (March, 26, 2020).
